### PR TITLE
Use sqlalchemy's 'quote' function to quote table names and fix table quoting in cfr exporter

### LINKED ISSUE #168: [infra] Improve `quote_table_name` to accept full-qualified table identifiers
> @seut: Do you think this routine needs to be improved? You mentioned something about "quoting going south". Maybe the root cause is here, because the routine may only handle a few situations correctly?

> Yes the issue here is that it will not quote anything if the ident contains a `.` like `foo.bar`. This looks a bit wrong as it normally should quote an ident which contains dots, this would be a valid table name for PG but not for CrateDB which forbids using a `.` inside a table identifier (see https://cratedb.com/docs/crate/reference/en/latest/general/ddl/create-table.html#naming-restrictions).
But then I wonder why you're not using [sqlalchemy.sql.expression.quoted_name](https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.sql.expression.quoted_name) which works similar afaik.
That it tries to detect a FQ identifier, splits it and quotes all parts separately, would be a bit custom but needed if the schema and table idents are not quoted dedicated.

_Originally posted by @seut in https://github.com/crate-workbench/cratedb-toolkit/pull/88#discussion_r1627267605_

### PR DESCRIPTION
**Pull request by seut**
See commits.

Closes #168.

### CODE REVIEW

**Comment by seut** [comment]
Moving back to draft as there seem to be issues.

**Comment by amotl** [comment]
> Moving back to draft as there seem to be issues.

It might be totally unrelated, but it also might be related to unfortunate side-effects of shortcomings within the test suite, as only three test cases are failing. Let me bring in this patch first, and then have another look after rebasing.

- https://github.com/crate-workbench/sqlalchemy-cratedb/pull/10

**Comment by seut** [comment]
> It might be totally unrelated, but it also might be related to unfortunate side-effects of shortcomings within the test suite, as only three test cases are failing. Let me bring in this patch first, and then have another look after rebasing.
> 
>     * [Dialect: Fix `get_table_names()` reflection method sqlalchemy-cratedb#10](https://github.com/crate-workbench/sqlalchemy-cratedb/pull/10)

Yes I think it's unrelated. This PR should not have issues anymore now.

**Review by amotl — APPROVED** [review]
Thank you!

File: tests/util/database.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+
+def test_quote_ident():
+    database = DatabaseAdapter(dburi="crate://localhost")
+    assert database.quote_ident("my_table") == "my_table"
+    assert database.quote_ident("my-table") == '"my-table"'
+    assert database.quote_ident("MyTable") == '"MyTable"'
+    assert database.quote_ident('"MyTable"') == '"MyTable"'
+    assert database.quote_ident("my_schema.my_table") == "my_schema.my_table"
+    assert database.quote_ident("my-schema.my_table") == '"my-schema".my_table'
```
Comment (amotl) [inline tests/util/database.py:11]: What about this case and friends?
```python
assert database.quote_ident("my_schema"."my_table") == "my_schema.my_table"
```

Comment (seut) [reply tests/util/database.py:11]: The current logic will not remove such valid quotes, why should it?

**Comment by seut** [comment]
I think we should rename the `quote_ident` function to `quote_relation_name` to express it's difference from DB's usual `quote_ident` functions.
As written at https://github.com/crate/sqlalchemy-cratedb/issues/69#issuecomment-2167394339, a `quote` or `quote_identifier` function is generally not able and intended to work on full-qualified (relation/column) names but only on the unqualified identifier part.

**Comment by seut** [comment]
@amotl 
I've added a sanity check at the quoting function and also renamed it to express better what it does, pls check.

**Review by amotl — APPROVED** [review]

**Pull request merged**